sklearn/utils/_encode.py

```diff
@@ -48,9 +66,85 @@ def _unique(values, *, return_inverse=False, return_counts=False):
     )
 
 
+def _unique_pandas(values, *, return_counts=False):
+    if return_counts:
+        value_counts = values.value_counts(dropna=False, sort=False)
+        # sort categorical columns in lexical order to be consistent with order
+        # obtained when sorting after conversion to numpy array
+        try:
+            value_counts.index = value_counts.index.reorder_categories(
+                value_counts.index.categories.sort_values()
+            )
+        except AttributeError:
+            pass
+        value_counts = value_counts.sort_index()
+        return value_counts.index.to_numpy(), value_counts.to_numpy()
+    unique = values.unique()
+    # unique returns a NumpyExtensionArray for extension dtypes and a numpy
+    # array for other dtypes
```

Comment:
Is this comment true? For this categorical dtype the output is not `NumpyExtensionArray` or a `np.ndarray`:

```python
import pandas as pd
from pandas.arrays import NumpyExtensionArray
import numpy as np

x = pd.Series(["a", "b", "c", "b", "a", "b"],dtype="category")
uniques = x.unique()

assert not isinstance(uniques, NumpyExtensionArray)
assert not isinstance(uniques, np.ndarray)
```